From d23246f35a23d099a25787841d149c8500a3d2cb Mon Sep 17 00:00:00 2001
From: Chris Morgan <macromorgan@hotmail.com>
Date: Wed, 30 Oct 2024 17:54:04 -0500
Subject: [PATCH] arm64: dts: rockchip: Add vcc for vibrator node on RGxx3

Define the vcc supply for the vibrator on the Anbernic RGxx3 series
devices. This supply value was discovered by enabling the pwm channel
to full duty and then slightly raising and lowering the voltages of
various regulators from within the bootloader and listening for
harmonic changes from the vibrator.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
index 67cb89162a89..b9018bed29de 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
@@ -144,6 +144,7 @@ vibrator: pwm-vibrator {
 		compatible = "pwm-vibrator";
 		pwm-names = "enable";
 		pwms = <&pwm5 0 1000000000 0>;
+		vcc-supply = <&vcc_3v3>;
 	};
 };
 
-- 
2.34.1

